src/figuresClasses.ts

```diff
@@ -1,11 +1,81 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: 'triangle' | 'circle' | 'rectangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = 'triangle';
+
+    const sumAB: number = this.a + this.b;
+    const sumAC: number = this.a + this.c;
+    const sumBC: number = this.b + this.c;
+
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('Error');
+    }
+
+    if (this.a >= sumAC || this.b >= sumBC || this.c >= sumAB) {
+      throw new Error('Error1');
```

Comment:
[CODE STYLE] - The error message should describe what the error is about. A better error message would be 'Triangle inequality violated: one side is too long.'